src/pages/FoodItem/FoodItemPage.tsx

```diff
@@ -49,47 +72,14 @@ export default function FoodItemPage() {
     },
   });
 
-  //update value of Total Price whenever form changes.
-  //issue if I directly use getValues('finalPrice')
-  const currentFormValues = methods.getValues("options");
-  const currentFormState = methods.getFieldState("options");
-  useEffect(() => {
-    let additionalCharge = 0;
-    foodItem.options.forEach((option, optionIndex) => {
-      const intersection = option.subOptions.filter((subOption) =>
-        currentFormValues[optionIndex].subOptions.includes(subOption.name)
-      );
-      option.subOptions.forEach((subOption) => {
-        if (intersection.includes(subOption))
-          additionalCharge += subOption.price;
-      });
-    });
-    const finalPrice = (foodItem.basePrice + additionalCharge) * quantity;
-    methods.setValue("quantity", quantity);
-    methods.setValue("finalPrice", finalPrice);
-    setTotalPrice(finalPrice);
-  }, [
-    foodItem,
-    setTotalPrice,
-    currentFormValues,
-    currentFormState,
-    methods,
-    quantity,
-  ]);
+  methods.setValue("finalPrice", priceState.totalPrice * quantity);
```

Comment:
What is this for?

Reply:
Update form state.
Set finalPrice in the form to the totalPrice * quantity.

Reply:
No, finalPrice is a derived attribute, not a user input.
The calculation is done at the backend? So should I remove it from the form field?

For UI, the total price displayed on the button, I just used quantityState * priceState, I don't need another variable to store this finalPrice.